aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -150,6 +187,167 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store {}
 
+    /// The mapping between coin and fungible asset.
+    struct CoinConversionMap has key {
+        coin_to_fungible_asset_map: Table<TypeInfo, address>,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The paired coin type info stored in fungible asset metadata object.
+    struct PairedCoinType has key {
+        type: TypeInfo,
```

Comment:
Why not use generic here?

Reply:
you mean `PairedCoinType<T>`?
cuz I have to query this struct to know which Coin it is paired with. before querying, you don't know what's T